src/main/java/picard/analysis/CollectWgsMetricsWithNonZeroCoverage.java

```diff
@@ -157,18 +156,18 @@ protected WgsMetrics generateWgsMetrics(final IntervalList intervals,
                 pctExcludedByCapping,
                 pctTotal,
                 coverageCap,
-                baseQHistogram,
+                unfilteredBaseQHistogram,
                 sampleSize);
     }
+
     @Override
     protected WgsMetricsCollector getCollector(final int coverageCap, final IntervalList intervals) {
-        assert(coverageCap == this.collector.coverageCap);
-        return this.collector;
+        if (collector != null) throw new PicardException("A WGSMetricsCollector already exists");
```

Comment:
@takutosato I am going to stick with my implementation as this only fixes one issue, while I found a few more when I fixed this and wrote some test.